### ISSUE TITLE
Installation problem on Debian 9 - with solution 

### ISSUE THREAD

**Issue by schloegl** [bcgsc/LINKS#16]
When trying to install links 1.8.6 on Debian 9 (Stretch), I observed the following problems. 

- try to use the binary failed with this error: 

```
LINKS-1.8.6/releases/links_v1.8.6$ ./LINKS
Can't load
'/localhome/schloegl/src/links-c.build/LINKS-1.8.6/releases/links_v1.8.6/./lib/bloomfilter/swig/BloomFilter.so'
for module BloomFilter:
/localhome/schloegl/src/links-c.build/LINKS-1.8.6/releases/links_v1.8.6/./lib/bloomfilter/swig/BloomFilter.so:
undefined symbol: PL_stack_sp at
/usr/lib/x86_64-linux-gnu/perl/5.24/DynaLoader.pm line 187.
 at
/localhome/schloegl/src/links-c.build/LINKS-1.8.6/releases/links_v1.8.6/./lib/bloomfilter/swig/BloomFilter.pm
line 11.
Compilation failed in require at ./LINKS line 26.
BEGIN failed--compilation aborted at ./LINKS line 26.

```

- try to compile links according to the instructions 
```  swig -Wall -c++ -perl5 BloomFilter.i
  g++ -c BloomFilter_wrap.cxx -I/usr/lib64/perl5/CORE -fPIC -Dbool=char -O3
  g++ -Wall -shared BloomFilter_wrap.o -o BloomFilter.so -O3
```

failed with 

```
In file included from /usr/include/c++/6/cmath:42:0,
                  from /usr/include/c++/6/math.h:36,
                  from BloomFilter_wrap.cxx:758:
/usr/include/c++/6/bits/cpp_type_traits.h:145:12: error: redefinition
of ‘struct std::__is_integer<char>’
      struct __is_integer<char>
             ^~~~~~~~~~~~~~~~~~
/usr/include/c++/6/bits/cpp_type_traits.h:138:12: error: previous
definition of ‘struct std::__is_integer<char>’
      struct __is_integer<bool>
             ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/6/exception:172:0,
                  from /usr/include/c++/6/stdexcept:38,
                  from BloomFilter_wrap.cxx:1568:
/usr/include/c++/6/bits/exception_ptr.h: In member function
‘std::__exception_ptr::exception_ptr::operator char() const’:
/usr/include/c++/6/bits/exception_ptr.h:143:16: error: invalid
conversion from ‘void*’ to ‘char’ [-fpermissive]
        { return _M_exception_object; }
                 ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/6/bits/move.h:57:0,
                  from /usr/include/c++/6/bits/nested_exception.h:40,
                  from /usr/include/c++/6/exception:173,
                  from /usr/include/c++/6/stdexcept:38,
                  from BloomFilter_wrap.cxx:1568:
/usr/include/c++/6/type_traits: At global scope:
/usr/include/c++/6/type_traits:224:12: error: redefinition of ‘struct
std::__is_integral_helper<char>’
      struct __is_integral_helper<char>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/type_traits:220:12: error: previous definition of
‘struct std::__is_integral_helper<char>’
      struct __is_integral_helper<bool>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/6/exception:173:0,
                  from /usr/include/c++/6/stdexcept:38,
                  from BloomFilter_wrap.cxx:1568:
/usr/include/c++/6/bits/nested_exception.h: In member function ‘void
std::nested_exception::rethrow_nested() const’:
/usr/include/c++/6/bits/nested_exception.h:75:11: error: invalid
user-defined conversion from ‘const
std::__exception_ptr::exception_ptr’ to ‘bool’ [-fpermissive]
        if (_M_ptr)
            ^~~~~~
In file included from /usr/include/c++/6/exception:172:0,
                  from /usr/include/c++/6/stdexcept:38,
                  from BloomFilter_wrap.cxx:1568:
/usr/include/c++/6/bits/exception_ptr.h:142:16: note: candidate is:
std::__exception_ptr::exception_ptr::operator char() const <near match>
        explicit operator bool() const
                 ^~~~~~~~
/usr/include/c++/6/bits/exception_ptr.h:142:16: note:   return type
‘char’ of explicit conversion function cannot be converted to ‘bool’
with a qualification conversion
In file included from /usr/include/c++/6/bits/basic_string.h:5643:0,
                  from /usr/include/c++/6/string:52,
                  from /usr/include/c++/6/stdexcept:39,
                  from BloomFilter_wrap.cxx:1568:
/usr/include/c++/6/bits/functional_hash.h: At global scope:
/usr/include/c++/6/bits/functional_hash.h:111:3: error: redefinition
of ‘struct std::hash<char>’
    _Cxx_hashtable_define_trivial_hash(char)
    ^
/usr/include/c++/6/bits/functional_hash.h:108:3: error: previous
definition of ‘struct std::hash<char>’
    _Cxx_hashtable_define_trivial_hash(bool)
    ^
In file included from /usr/include/c++/6/bits/uniform_int_dist.h:35:0,
                  from /usr/include/c++/6/bits/stl_algo.h:66,
                  from /usr/include/c++/6/algorithm:62,
                  from BloomFilter_wrap.cxx:1610:
/usr/include/c++/6/limits:451:12: error: redefinition of ‘struct
std::numeric_limits<char>’
      struct numeric_limits<char>
             ^~~~~~~~~~~~~~~~~~~~
/usr/include/c++/6/limits:382:12: error: previous definition of
‘struct std::numeric_limits<char>’
      struct numeric_limits<bool>
             ^~~~~~~~~~~~~~~~~~~~
```

the default gcc compiler in debian 9 is g++ v6.3

In order to address these, the following issues must be considered: 
- setting the proper perl confgure path, and 
- using g++-4.9 or 
- omitting the argument "-Dbool=char"

In summary, the following commands eabled the compilation on Debian 9: 
```
    swig -Wall -c++ -perl5 BloomFilter.i 
    g++ -c BloomFilter_wrap.cxx -I$(perl -e 'use Config; print $Config{archlib};')"/CORE" -fPIC -O3
    g++ -Wall -shared BloomFilter_wrap.o -o BloomFilter.so -O3`

```
BTW, "-Dbool=char" does not seem to be needed with gcc-4.9 either. 

I'd like suggesting to include this information in the documentation about the installation, so others do not need to figure this out again.

**Comment by rafaeltiveron** [comment]
Yes, It helped me. But, I needed to symlink bloomfilter.pm and associated files in CORE folder. Then, I run ./test.pl:

```
$ ./test.pl
de novo bf tests done
Storing filter. Filter is 125000000bytes.
Writting header... magic: BlOOMFXX hlen: 72 size: 1000000000 nhash: 5 kmer: 20 dFPR: 0 aFPR: 0 rFPR: 0 nEntry: 0 tEntry: 0
Loading header... magic: BlOOMFXX hlen: 72 size: 1000000000 nhash: 5 kmer: 20 dFPR: 0 aFPR: 0 rFPR: 0 nEntry: 0 tEntry: 0
premade bf tests done
Filter Info: Pop - 20, numHash - 5, kmerSize - 20, size - 1000000000
0 TAGAA found
1 AGAAT found
2 GAATC found
3 AATCA found
4 ATCAC found
5 TCACC found
6 CACCC found
7 ACCCA found
8 CCCAA found
9 CCAAA found
10 CAAAG found
11 AAAGA found
Done!
```

Does it indicate a successfull instalation?
